File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/CreatureRepository.java
package me.nathanp.magiccreatures.model;

import java.util.List;

import androidx.lifecycle.LiveData;

public class CreatureRepository {
    private static final String TAG = "CreatureRepository";
    private CreatureRemoteDao remoteDao;

    public CreatureRepository() {
        remoteDao = new CreatureRemoteDao();
    }

    public LiveData<List<Creature>> getAllCreatures() {
        return remoteDao.getAllCreatures();
    }

    public String addCreature(Creature creature) {
        return remoteDao.add(creature);
    }

    public void deleteCreature(String id) {
        remoteDao.delete(id);
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/CreatureRepository.kt
package me.nathanp.magiccreatures.model

import androidx.lifecycle.LiveData

class CreatureRepository {
    private val remoteDao = CreatureRemoteDao()
    val allCreatures: LiveData<List<Creature>>
        get() = remoteDao.allCreatures

    fun addCreature(creature: Creature): String {
        return remoteDao.add(creature)
    }

    fun deleteCreature(id: String) {
        remoteDao.delete(id)
    }
}